test/extensions/INTEL/SPV_INTEL_fpga_buffer_location/FPGABufferLocation.ll

```diff
@@ -30,20 +31,75 @@
 source_filename = "buffer_location.cl"
 target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
 target triple = "spir64-unknown-unknown"
 
+@.str.1 = private unnamed_addr constant [9 x i8] c"main.cpp\00", section "llvm.metadata"
+@.str.4 = private unnamed_addr constant [19 x i8] c"{5921:\22123456789\22}\00", section "llvm.metadata"
+
+; Function Attrs: norecurse nounwind readnone
+define spir_kernel void @test(i32 addrspace(1)* %a, float addrspace(1)* %b, i32 addrspace(1)* %c, i32 %d, i32 %e) local_unnamed_addr !kernel_arg_addr_space !3 !kernel_arg_access_qual !4 !kernel_arg_type !5 !kernel_arg_base_type !5 !kernel_arg_buffer_location !6
+; CHECK-LLVM: !kernel_arg_buffer_location ![[BUFLOC_MD:[0-9]+]]
+{
+entry:
+  ret void
+}
+
+; test1 : direct on kernel argument
 ; Function Attrs: norecurse nounwind readnone
-define spir_kernel void @test(i32 addrspace(1)* %a, float addrspace(1)* %b, i32 addrspace(1)* %c, i32 %d, i32 %e) local_unnamed_addr !kernel_arg_addr_space !3 !kernel_arg_access_qual !4 !kernel_arg_type !5 !kernel_arg_base_type !5 !kernel_arg_buffer_location !6 {
+define spir_kernel void @test.1(i8 addrspace(4)* %a) #0
+; CHECK-LLVM: !kernel_arg_buffer_location ![[BUFLOC_MD_TEST1:[0-9]+]]
+{
 entry:
+  %0 = call i8 addrspace(4)* @llvm.ptr.annotation.p4i8(i8 addrspace(4)* %a, i8* getelementptr inbounds ([19 x i8], [19 x i8]* @.str.4, i32 0, i32 0), i8* getelementptr inbounds ([9 x i8], [9 x i8]* @.str.1, i32 0, i32 0), i32 7, i8* null)
+  store i8 0, i8 addrspace(4)* %0, align 8
   ret void
 }
 
+%struct.MyIP = type { i32 addrspace(4)* }
+$test.2 = comdat any
+
+; test2 : general
+; Function Attrs: convergent mustprogress norecurse
+define weak_odr dso_local spir_kernel void @test.2(i32 addrspace(1)* noundef align 4 %arg_a) #0 comdat !kernel_arg_buffer_location !7 {
+entry:
+  %this.addr.i = alloca %struct.MyIP addrspace(4)*, align 8
+  %arg_a.addr = alloca i32 addrspace(1)*, align 8
+  %MyIP = alloca %struct.MyIP, align 8
+  %arg_a.addr.ascast = addrspacecast i32 addrspace(1)** %arg_a.addr to i32 addrspace(1)* addrspace(4)*
+  %MyIP.ascast = addrspacecast %struct.MyIP* %MyIP to %struct.MyIP addrspace(4)*
+  store i32 addrspace(1)* %arg_a, i32 addrspace(1)* addrspace(4)* %arg_a.addr.ascast, align 8
+  %0 = bitcast %struct.MyIP* %MyIP to i8*
+  %a = getelementptr inbounds %struct.MyIP, %struct.MyIP addrspace(4)* %MyIP.ascast, i32 0, i32 0
+  %1 = bitcast i32 addrspace(4)* addrspace(4)* %a to i8 addrspace(4)*
+  %2 = call i8 addrspace(4)* @llvm.ptr.annotation.p4i8(i8 addrspace(4)* %1, i8* getelementptr inbounds ([19 x i8], [19 x i8]* @.str.4, i32 0, i32 0), i8* getelementptr inbounds ([9 x i8], [9 x i8]* @.str.1, i32 0, i32 0), i32 7, i8* null)
+; CHEKC-LLVM: call i32 addrspace(4)* addrspace(4)* @llvm.ptr.annotation.p4p4i32.p0i8(i32 addrspace(4)* addrspace(4)* %a, i8* getelementptr inbounds ([33 x i8], [33 x i8]* @0, i32 0, i32 0), i8* undef, i32 undef, i8* undef)
```

Comment:
Would be nice to check the content of generated annotation strings (e.g., `@0`)

Reply:
added.